Wildcards aren't captured for inner classes that use type parameters from the outer class

```java
abstract class CaptureOuter<T> {
  abstract T get();

  abstract class Inner {
    abstract T get();
  }
}

class Client {
  Object getFrom(CaptureOuter<?> o) {
    return o.get();
  }

  Object getFrom(CaptureOuter<?>.Inner o) {
    return o.get();
  }
}
```

```
$ $HOME/checker-framework-3.21.0/checker/bin/javac -processor nullness CaptureOuter.java
CaptureOuter.java:11: error: [return] incompatible types in return.
    return o.get();
                ^
  type of expression: capture#84 extends @Initialized @Nullable Object
  method return type: @Initialized @NonNull Object
CaptureOuter.java:15: error: [return] incompatible types in return.
    return o.get();
                ^
  type of expression: @Initialized ? extends @Initialized @Nullable Object
  method return type: @Initialized @NonNull Object
2 errors
```
 
The call to CaptureOuter.get() has a type of `capture#84 extends @Initialized @Nullable Object`, as expected.
 
But the call to CaptureOuter.Inner.get() has a type of `@Initialized ? extends @Initialized @Nullable Object`.
 
Contrast this to what javac does if the return types of the getFrom methods is changed to String: In both cases, it identifies the call as having a type of `CAP#1`.

--- a/checker/tests/nullness/NullnessIssue4996.java
+++ b/checker/tests/nullness/NullnessIssue4996.java
@@ -9,10 +9,12 @@ class NullnessIssue4996 {
 
   class Client {
     Object getFrom(CaptureOuter<?> o) {
+      // :: error: (return)
       return o.get();
     }
 
     Object getFrom(CaptureOuter<?>.Inner o) {
+      // :: error: (return)
       return o.get();
     }
   }

--- a/checker/tests/nullness/NullnessIssue4996.java
+++ b/checker/tests/nullness/NullnessIssue4996.java
@@ -1,0 +1,19 @@
+class NullnessIssue4996 {
+  abstract class CaptureOuter<T> {
+    abstract T get();
+
+    abstract class Inner {
+      abstract T get();
+    }
+  }
+
+  class Client {
+    Object getFrom(CaptureOuter<?> o) {
+      return o.get();
+    }
+
+    Object getFrom(CaptureOuter<?>.Inner o) {
+      return o.get();
+    }
+  }
+}

--- a/framework/src/main/java/org/checkerframework/framework/util/AnnotatedTypes.java
+++ b/framework/src/main/java/org/checkerframework/framework/util/AnnotatedTypes.java
@@ -512,6 +512,7 @@ public class AnnotatedTypes {
     AnnotatedDeclaredType enclosingType = atypeFactory.getAnnotatedType(enclosingClassOfElem);
     AnnotatedDeclaredType base =
         (AnnotatedDeclaredType) asOuterSuper(types, atypeFactory, t, enclosingType);
+    base = (AnnotatedDeclaredType) atypeFactory.applyCaptureConversion(base);
 
     final List<AnnotatedTypeVariable> ownerParams =
         new ArrayList<>(enclosingType.getTypeArguments().size());

--- a/framework/tests/all-systems/Issue4996.java
+++ b/framework/tests/all-systems/Issue4996.java
@@ -1,0 +1,21 @@
+public class Issue4996 {
+
+  abstract static class CaptureOuter<T extends CharSequence> {
+    abstract T get();
+
+    abstract class Inner {
+      abstract T get();
+    }
+  }
+
+  static class Client {
+
+    CharSequence getFrom(CaptureOuter<?> o) {
+      return o.get();
+    }
+
+    CharSequence getFrom(CaptureOuter<?>.Inner i) {
+      return i.get();
+    }
+  }
+}

